scipy/optimize/_optimize.py

```diff
@@ -2130,6 +2130,8 @@ def terminate(warnflag, msg):
                                 njev=sf.ngev, nhev=hcalls, status=warnflag,
                                 success=(warnflag == 0), message=msg, x=xk,
                                 nit=k)
+        if fhess is not None:
```

Comment:
5. I don't know.
6. I won't. I am just closing this PR. Thank you.

Reply:
Actually, what I don't understand here is the attribution - it looks like A is not defined here. It might be that we want the other way around (`A = result.hess`?) , but from skimming the code I think we may not need these two lines (tests pass withouth them)